### PR TITLE
feat: refined return type of DB::transaction method

### PR DESCRIPTION
**Pull request by jdjfisher**
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

Refines the return type of the `\Illuminate\Support\Facades\DB::transaction` method.

### CODE REVIEW

**Comment by jdjfisher** [comment]
It would be great to also do something similar for `Cache::remember`php  as I find myself having to explicitly annotate the return type for that quite often. That said, I'm not sure if its possible to refine it while being entirely true to the implementation. e.g

```php
Cache::put('the-key', 'not-an-int');

$probably_an_int_ = Cache::rememberForever('the-key', fn () => rand());
```

Thoughts?

File: stubs/Database/ManagesTransactions.stub [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+trait ManagesTransactions
+{
+    /**
+     * Execute a Closure within a transaction.
+     *
+     * @template TCallbackReturnType
+     *
+     * @param  \Closure($this): TCallbackReturnType  $callback
+     * @param  int  $attempts
```
Comment (mad-briller) [inline stubs/Database/ManagesTransactions.stub:13]: You could make this positive-int as making 0 attempts makes no sense

Comment (jdjfisher) [reply stubs/Database/ManagesTransactions.stub:13]: Good idea 👍

**Review by szepeviktor — APPROVED** [review]

**Review by canvural — APPROVED** [review]

**Comment by canvural** [comment]
Thank you!

**Pull request merged**